com.unity.sg2/Editor/GraphUI/ConstantEditorFactoryExtensions.cs

```diff
@@ -52,7 +52,19 @@ public static BaseModelPropertyField BuildGraphTypeConstantEditor(this ConstantE
             {
                 case GraphDataPortModel graphDataPort:
                 {
-                    ((GraphDataNodeModel)graphDataPort.NodeModel).TryGetNodeHandler(out var nodeReader);
+                    NodeHandler nodeReader;
+
+                    switch (graphDataPort.NodeModel)
+                    {
+                        case GraphDataNodeModel graphDataNode:
```

Comment:
Some of this could be simplified down to remove references to CLDS now that we have the UI-side view model data, does it seem within scope to apply those changes within this PR so we don't need to go back and fix it up later?